src/content/login/Controller.tsx

```diff
@@ -9,12 +10,14 @@ const loginController = () => {
   const NIH_AUTHORIZE_URL = env.REACT_APP_NIH_AUTHORIZE_URL || env.NIH_AUTHORIZE_URL;
   const NIH_CLIENT_ID = env.REACT_APP_NIH_CLIENT_ID || env.NIH_CLIENT_ID;
   const NIH_REDIRECT_URL = env.REACT_APP_NIH_REDIRECT_URL || env.NIH_REDIRECT_URL;
-
+  const { state } = useLocation();
+  const redirectURLOnLoginSuccess = state && state.redirectURLOnLoginSuccess ? state.redirectURLOnLoginSuccess : null;
   const urlParam = {
     client_id: `${NIH_CLIENT_ID}`,
     redirect_uri: `${NIH_REDIRECT_URL}`,
     response_type: 'code',
-    scope: 'openid email profile'
+    scope: 'openid email profile',
+    state: redirectURLOnLoginSuccess
```

Comment:
Can we add the state param only if it's not null? When logging in without using the dialog, we're getting redirected to `[...url...]/null` which results in 404.